File: pattern.py
n=int(input("Enter any number > "))
m=n
a=[]
for i in range(2*n-1):
 a.append([])

for i in range(2*n-1):
    for j in range(2*n-1):
        a[i].append(0) 
for i in range(n):
    for j in range(i,2*n-1-i):
        for k in range(i,2*n-1-i):
            a[j][k]=m
    m-=1    

for i in range(2*n-1):
    print(a[i])

File: pattern.c
#include<stdio.h>

void main(){
    int a[15][15],n,m,i,j,k;

    printf("Enter any number > ");
    scanf("%d",&n);
    m=n;
    
    for(i=0;i<n;i++){
    for(j=i;j<2*n-1-i;j++)
        for(k=i;k<2*n-1-i;k++)
            a[j][k]=m;
    m--; 
    }
    
    for(i=0;i<2*n-1;i++){
     for (j=0;j<2*n-1;j++)
        printf("%d ",a[i][j]);
    printf("\n");
    }
           
}